src/vmm/src/logger/metrics.rs

```diff
@@ -78,6 +78,26 @@ use crate::devices::virtio::vhost_user_metrics;
 use crate::devices::virtio::virtio_block::metrics as block_metrics;
 use crate::devices::virtio::vsock::metrics as vsock_metrics;
 
+#[macro_export]
+/// records aggregate (min/max/sum) for the given $metrics
+/// This macro captures delta between start_time and current time
+/// and updates min/max/sum metrics so it expects that start_time
+/// is captured at the appropriate location.
+macro_rules! record_latency_summary {
+    ($metrics:expr, $start_time:ident) => {
+        let delta_us = utils::time::get_time_us(utils::time::ClockType::Monotonic) - $start_time;
+        $metrics.sum_us.add(delta_us);
+        let min_us = $metrics.min_us.fetch();
+        let max_us = $metrics.max_us.fetch();
+        if (0 == min_us) || (min_us > delta_us) {
+            $metrics.min_us.store(delta_us);
+        }
+        if (0 == max_us) || (max_us < delta_us) {
+            $metrics.max_us.store(delta_us);
+        }
+    };
+}
```

Comment:
Could this be
```suggestion
impl LatencyAggregateMetrics {
    pub fn record_latency_summary(&mut self, start_time: u64) {
        let delta_us = utils::time::get_time_us(utils::time::ClockType::Monotonic) - start_time;
        self.sum_us.add(delta_us);
        let min_us = self.min_us.fetch();
        let max_us = self.max_us.fetch();
        if (0 == min_us) || (min_us > delta_us) {
            self.min_us.store(delta_us);
        }
        if (0 == max_us) || (max_us < delta_us) {
            self.max_us.store(delta_us);
        }
    }
}
```

Reply:
Thanks for the suggestion, updated the code accordingly